src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_VALUE = 1.5;
+    private T[] storage;
+    private int size;
+
+    public ArrayList() {
+        storage = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == storage.length) {
+            growCapacity((int) (size * GROW_VALUE));
+        }
+        storage[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds");
+        }
+        if (size == storage.length) {
+            growCapacity((int) (size * GROW_VALUE));
+        }
+        System.arraycopy(storage, index, storage, index + 1, size - index);
+        storage[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int requiredCapacity = list.size() + size;
+        if (requiredCapacity > storage.length) {
+            growCapacity(requiredCapacity);
+        }
+        for (int i = 0; i < list.size(); i++) {
+            storage[size + i] = list.get(i);
+        }
+        size += list.size();
```

Comment:
too complex logic, let's go through all list and just call add method that you implemented above